lib/pure/os.nim

```diff
@@ -2051,8 +2055,8 @@ proc moveFile*(source, dest: string) {.rtl, extern: "nos$1",
   ## * `removeFile proc <#removeFile,string>`_
   ## * `tryRemoveFile proc <#tryRemoveFile,string>`_
 
-  if not tryMoveFSObject(source, dest):
-    when not defined(windows):
+  if not tryMoveFSObject(source, dest, isMoveDir=false):
+    when defined(windows):
```

Comment:
why change to `when defined(windows):` ? seems like a bug, it'll silently do nothing on posix when this is reached. instead how about:
```nim
when defined(windows):
  doAssert false
    # this points to a stdlib bug, not user error, on windows it should move
    # the file even if across partitions
else:
  # Fallback to copy & del
  ...
```

Reply:
whoops, yeah, sorry